lib/sneaky-save.rb

```diff
@@ -66,24 +66,45 @@ def sneaky_update
       # Here we have changes --> save them.
       pk = self.class.primary_key
       original_id = changed_attributes.has_key?(pk) ? changes[pk].first : send(pk)
-      !self.class.where(pk => original_id).update_all(attributes).zero?
+
+      changed_values = sneaky_update_fields
+
+      # serialize values for rails3 before updating.
+      s_f = self.class.serialized_attributes.keys & changed_values.keys
+      s_f.each_with_object(changed_values) do |key, val|
+        val[key] = @attributes[key].serialized_value
+      end unless rails4?
+
+      !self.class.where(pk => original_id).update_all(changed_values).zero?
     end
 
-    def skeaky_attributes_values
-      if ActiveRecord::VERSION::STRING.split('.').first.to_i > 3
-        send :arel_attributes_with_values_for_create, attribute_names
+    def sneaky_attributes_values
+      if rails4?
+        send(:arel_attributes_with_values_for_create, attribute_names)
       else
-        send :arel_attributes_values
+        send(:arel_attributes_values)
+      end
+    end
+
+    def sneaky_update_fields
+      changes.keys.each_with_object({}) do |field, value|
+        # do not trust values from changes directly
+        # overridden reader/writer methods might make it behave unexpectedly
+        value[field] = read_attribute(field)
       end
     end
 
     def sneaky_connection
-      if ActiveRecord::VERSION::STRING.split('.').first.to_i > 3
+      if rails4?
         self.class.connection
       else
         connection
       end
     end
+
+    def rails4?
```

Comment:
this method name is too generic, since we are mixin in the module into ar::Base globally. Please add sneaky prefix to it.